concent_api/core/message_handlers.py

```diff
@@ -956,34 +957,33 @@ def store_subtask(
 def handle_messages_from_database(client_public_key: bytes) -> Union[message.Message, None]:
     assert client_public_key not in ['', None]
     encoded_client_public_key = b64encode(client_public_key)
-
     with non_nesting_atomic(using='control'):
-        pending_response = PendingResponse.objects.select_for_update().filter(
+        payments_info_responses = PendingResponse.objects.select_for_update().filter(
+            payment_info__isnull=False,
             client__public_key=encoded_client_public_key,
             delivered=False,
         ).order_by('created_at').first()
 
-        if pending_response is None:
-            return None
+        subtask_responses = PendingResponse.objects.select_for_update().filter(
+            subtask__protocol_version=get_major_and_minor_golem_messages_version(settings.GOLEM_MESSAGES_VERSION),
+            client__public_key=encoded_client_public_key,
+            delivered=False,
+        ).order_by('created_at').first()
+        if payments_info_responses is not None and subtask_responses is not None:
```

Comment:
Could you please explain why we differentiate logic for `PendingResponse` with `payment_info` and without?

Reply:
@rwrzesien We need to differentiate because `PaymentInfo` does not contain `subtask_id` and we can't sort it by GolemMessages version which is stored in database. Also `PaymentInfo` is not dependent on the GolemMessages version becuase all information are stored in separate columns in database so we can create and serialize message in `2.24` and '3.2' GM version. Unless construction of `ForcePaymentComitted` will change.